 # Description

In this notebook I have collected data from a dedicated API: [click](https://api-datalab.coderslab.com/v2/docs/).

 According to the documentation there are 4 endpoints available:
 - `airport` - airport data, parameters: airport_id (the list of airports to be passed is in the **airports.csv** file in folder 'data' of project repository,
 - `weather` - weather details at the airport on a given day,
 - `aircraft` - aircrafts' manufacture data 
 - `flights` - daily data regarding departures from the airport.

### Remarks:

- To collect succesfully the data from these endpoints an authorization token is required. The API key is stored in the environment variable and is represented in the code as 'APIKEY'
- The construction of the endpoints differs from endpoint to endpoint requiring slightly different approaches to successfully collect all data.
- API call limit is 1000/min

 # Notebook configuration

In [1]:
import os
from dotenv import load_dotenv
import json
import requests
from time import sleep
from tqdm import tqdm
import pandas as pd

In [2]:
load_dotenv()
APIKEY = os.environ.get('API_KEY')
API_URL = 'https://api-datalab.coderslab.com/api/v2'

In [3]:
df = pd.read_csv(r'..\data\airports.csv')

In [4]:
df.shape

(364, 1)

In [5]:
df.head()

origin_airport_id
0              10874
1              11233
2              13360
3              15008
4              11638

In [6]:
airport_id_list = df['origin_airport_id'].unique().tolist()
len(airport_id_list)

364

There are 364 unique airports on the list

 # Collect data from `Airport` endpoint

In [7]:
airports = []

for airport in tqdm(airport_id_list):
    response_airport = requests.get(f'{API_URL}/airport/{airport}', headers={'Authorization': APIKEY})
    
    if response_airport.status_code == 200:    
        airports.append(response_airport.json())
    elif response_airport.status_code == 400:
        print(f'No data for airport {airport} (code status: 400)')
    else:
        print(f'Error retrieving airport data {airport} (code status: {response_airport.status_code})')

  0%|▏                                                                                 | 1/364 [00:00<05:06,  1.19it/s]

No data for airport 10874 (code status: 400)


  1%|▍                                                                                 | 2/364 [00:01<04:55,  1.22it/s]

No data for airport 11233 (code status: 400)


  1%|▋                                                                                 | 3/364 [00:02<04:42,  1.28it/s]

No data for airport 13360 (code status: 400)


  1%|▉                                                                                 | 4/364 [00:03<04:36,  1.30it/s]

No data for airport 15008 (code status: 400)


  2%|█▎                                                                                | 6/364 [00:04<04:30,  1.32it/s]

No data for airport 14150 (code status: 400)


  2%|█▌                                                                                | 7/364 [00:05<04:32,  1.31it/s]

No data for airport 15323 (code status: 400)


  2%|█▊                                                                                | 8/364 [00:06<04:38,  1.28it/s]

No data for airport 14814 (code status: 400)


  2%|██                                                                                | 9/364 [00:06<04:33,  1.30it/s]

No data for airport 12007 (code status: 400)


  3%|██▏                                                                              | 10/364 [00:07<04:30,  1.31it/s]

No data for airport 11337 (code status: 400)


  3%|██▋                                                                              | 12/364 [00:09<04:27,  1.31it/s]

No data for airport 15070 (code status: 400)


  4%|███                                                                              | 14/364 [00:10<04:35,  1.27it/s]

No data for airport 12280 (code status: 400)


  4%|███▌                                                                             | 16/364 [00:12<04:38,  1.25it/s]

No data for airport 11641 (code status: 400)


  5%|███▊                                                                             | 17/364 [00:13<04:34,  1.26it/s]

No data for airport 13832 (code status: 400)


  5%|████                                                                             | 18/364 [00:14<04:32,  1.27it/s]

No data for airport 10268 (code status: 400)


  5%|████▍                                                                            | 20/364 [00:15<04:24,  1.30it/s]

No data for airport 15041 (code status: 400)


  6%|████▉                                                                            | 22/364 [00:17<04:15,  1.34it/s]

No data for airport 12119 (code status: 400)


  6%|█████                                                                            | 23/364 [00:17<04:16,  1.33it/s]

No data for airport 11537 (code status: 400)


  7%|█████▎                                                                           | 24/364 [00:18<04:13,  1.34it/s]

No data for airport 11092 (code status: 400)


  7%|█████▌                                                                           | 25/364 [00:19<04:11,  1.35it/s]

No data for airport 10581 (code status: 400)


  7%|█████▊                                                                           | 26/364 [00:19<04:10,  1.35it/s]

No data for airport 13829 (code status: 400)


  7%|██████                                                                           | 27/364 [00:20<04:10,  1.34it/s]

No data for airport 15389 (code status: 400)


  8%|██████▍                                                                          | 29/364 [00:22<04:06,  1.36it/s]

No data for airport 12389 (code status: 400)


  8%|██████▋                                                                          | 30/364 [00:22<04:08,  1.35it/s]

No data for airport 11648 (code status: 400)


  9%|██████▉                                                                          | 31/364 [00:23<04:06,  1.35it/s]

No data for airport 15023 (code status: 400)


  9%|███████                                                                          | 32/364 [00:24<04:13,  1.31it/s]

No data for airport 11982 (code status: 400)


  9%|███████▎                                                                         | 33/364 [00:25<04:15,  1.30it/s]

No data for airport 10967 (code status: 400)


  9%|███████▌                                                                         | 34/364 [00:26<04:15,  1.29it/s]

No data for airport 11525 (code status: 400)


 10%|████████                                                                         | 36/364 [00:27<04:14,  1.29it/s]

No data for airport 14259 (code status: 400)


 10%|████████▏                                                                        | 37/364 [00:28<04:15,  1.28it/s]

No data for airport 11637 (code status: 400)


 10%|████████▍                                                                        | 38/364 [00:29<04:19,  1.26it/s]

No data for airport 10466 (code status: 400)


 11%|████████▉                                                                        | 40/364 [00:30<04:12,  1.28it/s]

No data for airport 10208 (code status: 400)


 11%|█████████                                                                        | 41/364 [00:31<04:12,  1.28it/s]

No data for airport 15841 (code status: 400)


 12%|█████████▌                                                                       | 43/364 [00:33<04:06,  1.30it/s]

No data for airport 12898 (code status: 400)


 12%|█████████▊                                                                       | 44/364 [00:33<04:11,  1.27it/s]

No data for airport 13241 (code status: 400)


 12%|██████████                                                                       | 45/364 [00:34<04:15,  1.25it/s]

No data for airport 13367 (code status: 400)


 13%|██████████▏                                                                      | 46/364 [00:35<04:11,  1.26it/s]

No data for airport 11481 (code status: 400)


 13%|██████████▍                                                                      | 47/364 [00:37<05:59,  1.13s/it]

No data for airport 14108 (code status: 400)


 13%|██████████▉                                                                      | 49/364 [00:39<05:17,  1.01s/it]

No data for airport 13873 (code status: 400)


 14%|███████████▏                                                                     | 50/364 [00:40<05:05,  1.03it/s]

No data for airport 10157 (code status: 400)


 14%|███████████▎                                                                     | 51/364 [00:40<04:48,  1.09it/s]

No data for airport 10245 (code status: 400)


 14%|███████████▌                                                                     | 52/364 [00:41<04:35,  1.13it/s]

No data for airport 11146 (code status: 400)


 15%|███████████▊                                                                     | 53/364 [00:42<04:20,  1.19it/s]

No data for airport 13277 (code status: 400)


 15%|████████████▏                                                                    | 55/364 [00:44<04:12,  1.22it/s]

No data for airport 11109 (code status: 400)


 15%|████████████▍                                                                    | 56/364 [00:44<04:06,  1.25it/s]

No data for airport 13459 (code status: 400)


 16%|████████████▋                                                                    | 57/364 [00:45<04:02,  1.27it/s]

No data for airport 11775 (code status: 400)


 16%|████████████▉                                                                    | 58/364 [00:46<03:59,  1.28it/s]

No data for airport 16218 (code status: 400)


 16%|█████████████▏                                                                   | 59/364 [00:47<03:55,  1.29it/s]

No data for airport 14698 (code status: 400)


 16%|█████████████▎                                                                   | 60/364 [00:47<03:51,  1.31it/s]

No data for airport 14252 (code status: 400)


 17%|█████████████▌                                                                   | 61/364 [00:48<03:51,  1.31it/s]

No data for airport 13256 (code status: 400)


 17%|█████████████▊                                                                   | 62/364 [00:49<03:49,  1.32it/s]

No data for airport 13139 (code status: 400)


 17%|██████████████                                                                   | 63/364 [00:50<03:49,  1.31it/s]

No data for airport 12250 (code status: 400)


 18%|██████████████▍                                                                  | 65/364 [00:51<03:43,  1.34it/s]

No data for airport 11468 (code status: 400)


 18%|██████████████▋                                                                  | 66/364 [00:52<03:41,  1.34it/s]

No data for airport 14952 (code status: 400)


 18%|██████████████▉                                                                  | 67/364 [00:53<03:42,  1.33it/s]

No data for airport 12402 (code status: 400)


 19%|███████████████▏                                                                 | 68/364 [00:53<03:41,  1.34it/s]

No data for airport 14574 (code status: 400)


 19%|███████████████▌                                                                 | 70/364 [00:55<03:43,  1.32it/s]

No data for airport 11977 (code status: 400)


 20%|███████████████▊                                                                 | 71/364 [00:56<03:43,  1.31it/s]

No data for airport 11867 (code status: 400)


 20%|████████████████                                                                 | 72/364 [00:56<03:46,  1.29it/s]

No data for airport 11203 (code status: 400)


 21%|████████████████▋                                                                | 75/364 [00:59<03:42,  1.30it/s]

No data for airport 10747 (code status: 400)


 21%|████████████████▉                                                                | 76/364 [01:00<03:40,  1.31it/s]

No data for airport 14905 (code status: 400)


 21%|█████████████████▏                                                               | 77/364 [01:00<03:36,  1.33it/s]

No data for airport 12012 (code status: 400)


 21%|█████████████████▎                                                               | 78/364 [01:01<03:33,  1.34it/s]

No data for airport 14783 (code status: 400)


 22%|█████████████████▊                                                               | 80/364 [01:02<03:27,  1.37it/s]

No data for airport 10431 (code status: 400)


 22%|██████████████████                                                               | 81/364 [01:03<03:29,  1.35it/s]

No data for airport 10434 (code status: 400)


 23%|██████████████████▏                                                              | 82/364 [01:04<03:29,  1.34it/s]

No data for airport 16869 (code status: 400)


 23%|██████████████████▍                                                              | 83/364 [01:05<03:52,  1.21it/s]

No data for airport 10408 (code status: 400)


 24%|███████████████████▊                                                             | 89/364 [01:10<03:36,  1.27it/s]

No data for airport 11624 (code status: 400)


 25%|████████████████████                                                             | 90/364 [01:10<03:42,  1.23it/s]

No data for airport 13541 (code status: 400)


 25%|████████████████████▎                                                            | 91/364 [01:11<03:49,  1.19it/s]

No data for airport 13422 (code status: 400)


 26%|█████████████████████▏                                                           | 95/364 [01:15<03:42,  1.21it/s]

No data for airport 14689 (code status: 400)


 26%|█████████████████████▎                                                           | 96/364 [01:15<03:39,  1.22it/s]

No data for airport 12391 (code status: 400)


 27%|█████████████████████▌                                                           | 97/364 [01:16<03:35,  1.24it/s]

No data for airport 10868 (code status: 400)


 27%|█████████████████████▊                                                           | 98/364 [01:17<03:35,  1.23it/s]

No data for airport 14711 (code status: 400)


 27%|█████████████████████▉                                                          | 100/364 [01:19<03:30,  1.25it/s]

No data for airport 11067 (code status: 400)


 28%|██████████████████████▏                                                         | 101/364 [01:19<03:25,  1.28it/s]

No data for airport 10562 (code status: 400)


 28%|██████████████████████▍                                                         | 102/364 [01:20<03:22,  1.29it/s]

No data for airport 11695 (code status: 400)


 29%|██████████████████████▊                                                         | 104/364 [01:22<03:20,  1.30it/s]

No data for airport 14109 (code status: 400)


 29%|███████████████████████                                                         | 105/364 [01:22<03:17,  1.31it/s]

No data for airport 13970 (code status: 400)


 29%|███████████████████████▌                                                        | 107/364 [01:24<03:17,  1.30it/s]

No data for airport 11076 (code status: 400)


 30%|███████████████████████▋                                                        | 108/364 [01:25<03:19,  1.29it/s]

No data for airport 14092 (code status: 400)


 30%|███████████████████████▉                                                        | 109/364 [01:25<03:19,  1.28it/s]

No data for airport 11122 (code status: 400)


 30%|████████████████████████▏                                                       | 110/364 [01:26<03:14,  1.30it/s]

No data for airport 11288 (code status: 400)


 30%|████████████████████████▍                                                       | 111/364 [01:27<03:20,  1.26it/s]

No data for airport 11308 (code status: 400)


 31%|████████████████████████▌                                                       | 112/364 [01:28<03:20,  1.26it/s]

No data for airport 10754 (code status: 400)


 31%|████████████████████████▊                                                       | 113/364 [01:29<03:24,  1.23it/s]

No data for airport 12884 (code status: 400)


 32%|█████████████████████████▎                                                      | 115/364 [01:30<03:19,  1.25it/s]

No data for airport 14588 (code status: 400)


 32%|█████████████████████████▋                                                      | 117/364 [01:32<03:13,  1.28it/s]

No data for airport 12915 (code status: 400)


 33%|██████████████████████████▎                                                     | 120/364 [01:34<03:04,  1.32it/s]

No data for airport 11603 (code status: 400)


 33%|██████████████████████████▌                                                     | 121/364 [01:35<03:03,  1.32it/s]

No data for airport 14457 (code status: 400)


 34%|██████████████████████████▊                                                     | 122/364 [01:36<03:03,  1.32it/s]

No data for airport 12206 (code status: 400)


 34%|███████████████████████████                                                     | 123/364 [01:36<03:03,  1.31it/s]

No data for airport 11252 (code status: 400)


 34%|███████████████████████████▎                                                    | 124/364 [01:37<03:03,  1.31it/s]

No data for airport 11905 (code status: 400)


 35%|███████████████████████████▋                                                    | 126/364 [01:39<03:04,  1.29it/s]

No data for airport 14120 (code status: 400)


 35%|███████████████████████████▉                                                    | 127/364 [01:40<03:03,  1.29it/s]

No data for airport 11980 (code status: 400)


 35%|████████████████████████████▏                                                   | 128/364 [01:40<03:01,  1.30it/s]

No data for airport 14025 (code status: 400)


 35%|████████████████████████████▎                                                   | 129/364 [01:41<03:00,  1.30it/s]

No data for airport 11150 (code status: 400)


 36%|████████████████████████████▌                                                   | 130/364 [01:42<02:58,  1.31it/s]

No data for airport 14709 (code status: 400)


 36%|████████████████████████████▊                                                   | 131/364 [01:43<02:55,  1.32it/s]

No data for airport 15897 (code status: 400)


 37%|█████████████████████████████▏                                                  | 133/364 [01:44<02:53,  1.33it/s]

No data for airport 14006 (code status: 400)


 37%|█████████████████████████████▋                                                  | 135/364 [01:46<02:53,  1.32it/s]

No data for airport 12902 (code status: 400)


 37%|█████████████████████████████▉                                                  | 136/364 [01:46<02:52,  1.32it/s]

No data for airport 13061 (code status: 400)


 38%|██████████████████████████████                                                  | 137/364 [01:47<02:50,  1.33it/s]

No data for airport 12016 (code status: 400)


 38%|██████████████████████████████▎                                                 | 138/364 [01:48<02:51,  1.32it/s]

No data for airport 14794 (code status: 400)


 38%|██████████████████████████████▌                                                 | 139/364 [01:49<02:53,  1.30it/s]

No data for airport 11921 (code status: 400)


 38%|██████████████████████████████▊                                                 | 140/364 [01:49<02:53,  1.29it/s]

No data for airport 10731 (code status: 400)


 39%|██████████████████████████████▉                                                 | 141/364 [01:50<02:51,  1.30it/s]

No data for airport 14828 (code status: 400)


 39%|███████████████████████████████▍                                                | 143/364 [01:52<02:49,  1.30it/s]

No data for airport 12441 (code status: 400)


 40%|███████████████████████████████▋                                                | 144/364 [01:52<02:49,  1.30it/s]

No data for airport 14802 (code status: 400)


 40%|███████████████████████████████▊                                                | 145/364 [01:53<02:47,  1.31it/s]

No data for airport 13230 (code status: 400)


 40%|████████████████████████████████                                                | 146/364 [01:54<02:48,  1.29it/s]

No data for airport 10631 (code status: 400)


 41%|████████████████████████████████▌                                               | 148/364 [01:56<02:45,  1.30it/s]

No data for airport 10141 (code status: 400)


 41%|█████████████████████████████████▏                                              | 151/364 [01:58<02:45,  1.29it/s]

No data for airport 13184 (code status: 400)


 42%|█████████████████████████████████▍                                              | 152/364 [01:59<02:43,  1.29it/s]

No data for airport 10643 (code status: 400)


 42%|█████████████████████████████████▋                                              | 153/364 [01:59<02:41,  1.30it/s]

No data for airport 10333 (code status: 400)


 42%|█████████████████████████████████▊                                              | 154/364 [02:00<02:39,  1.32it/s]

No data for airport 12255 (code status: 400)


 43%|██████████████████████████████████                                              | 155/364 [02:01<02:42,  1.29it/s]

No data for airport 14487 (code status: 400)


 43%|██████████████████████████████████▎                                             | 156/364 [02:02<02:40,  1.30it/s]

No data for airport 12129 (code status: 400)


 43%|██████████████████████████████████▌                                             | 157/364 [02:03<02:41,  1.28it/s]

No data for airport 10551 (code status: 400)


 44%|██████████████████████████████████▉                                             | 159/364 [02:04<02:36,  1.31it/s]

No data for airport 14256 (code status: 400)


 44%|███████████████████████████████████▍                                            | 161/364 [02:06<02:40,  1.26it/s]

No data for airport 13377 (code status: 400)


 45%|███████████████████████████████████▊                                            | 163/364 [02:07<02:35,  1.29it/s]

No data for airport 10627 (code status: 400)


 45%|████████████████████████████████████                                            | 164/364 [02:08<02:36,  1.28it/s]

No data for airport 12335 (code status: 400)


 45%|████████████████████████████████████▎                                           | 165/364 [02:09<02:34,  1.28it/s]

No data for airport 14633 (code status: 400)


 46%|████████████████████████████████████▍                                           | 166/364 [02:09<02:32,  1.30it/s]

No data for airport 11274 (code status: 400)


 46%|████████████████████████████████████▋                                           | 167/364 [02:10<02:32,  1.30it/s]

No data for airport 14543 (code status: 400)


 46%|█████████████████████████████████████▏                                          | 169/364 [02:12<02:33,  1.27it/s]

No data for airport 10158 (code status: 400)


 47%|█████████████████████████████████████▎                                          | 170/364 [02:13<02:33,  1.27it/s]

No data for airport 12951 (code status: 400)


 47%|█████████████████████████████████████▌                                          | 171/364 [02:13<02:29,  1.29it/s]

No data for airport 14004 (code status: 400)


 47%|█████████████████████████████████████▊                                          | 172/364 [02:14<02:28,  1.30it/s]

No data for airport 12544 (code status: 400)


 48%|██████████████████████████████████████                                          | 173/364 [02:15<02:26,  1.30it/s]

No data for airport 10739 (code status: 400)


 48%|██████████████████████████████████████▏                                         | 174/364 [02:16<02:27,  1.29it/s]

No data for airport 10165 (code status: 400)


 48%|██████████████████████████████████████▍                                         | 175/364 [02:16<02:24,  1.31it/s]

No data for airport 13964 (code status: 400)


 48%|██████████████████████████████████████▋                                         | 176/364 [02:17<02:24,  1.30it/s]

No data for airport 14314 (code status: 400)


 49%|██████████████████████████████████████▉                                         | 177/364 [02:18<02:27,  1.27it/s]

No data for airport 10990 (code status: 400)


 49%|███████████████████████████████████████                                         | 178/364 [02:19<02:25,  1.28it/s]

No data for airport 14812 (code status: 400)


 49%|███████████████████████████████████████▌                                        | 180/364 [02:20<02:20,  1.31it/s]

No data for airport 10561 (code status: 400)


 50%|███████████████████████████████████████▊                                        | 181/364 [02:21<02:21,  1.29it/s]

No data for airport 10146 (code status: 400)


 50%|████████████████████████████████████████▏                                       | 183/364 [02:23<02:17,  1.31it/s]

No data for airport 15991 (code status: 400)


 51%|████████████████████████████████████████▋                                       | 185/364 [02:24<02:17,  1.31it/s]

No data for airport 15401 (code status: 400)


 51%|████████████████████████████████████████▉                                       | 186/364 [02:25<02:15,  1.31it/s]

No data for airport 11587 (code status: 400)


 51%|█████████████████████████████████████████                                       | 187/364 [02:26<02:14,  1.32it/s]

No data for airport 14520 (code status: 400)


 52%|█████████████████████████████████████████▌                                      | 189/364 [02:27<02:16,  1.28it/s]

No data for airport 10185 (code status: 400)


 52%|█████████████████████████████████████████▉                                      | 191/364 [02:29<02:11,  1.32it/s]

No data for airport 14534 (code status: 400)


 53%|██████████████████████████████████████████▋                                     | 194/364 [02:31<02:10,  1.30it/s]

No data for airport 11111 (code status: 400)


 54%|██████████████████████████████████████████▊                                     | 195/364 [02:32<02:11,  1.29it/s]

No data for airport 15024 (code status: 400)


 54%|███████████████████████████████████████████                                     | 196/364 [02:33<02:10,  1.29it/s]

No data for airport 12917 (code status: 400)


 54%|███████████████████████████████████████████▎                                    | 197/364 [02:33<02:10,  1.28it/s]

No data for airport 12124 (code status: 400)


 54%|███████████████████████████████████████████▌                                    | 198/364 [02:34<02:10,  1.27it/s]

No data for airport 12523 (code status: 400)


 55%|████████████████████████████████████████████▏                                   | 201/364 [02:37<02:06,  1.29it/s]

No data for airport 11630 (code status: 400)


 55%|████████████████████████████████████████████▍                                   | 202/364 [02:37<02:04,  1.30it/s]

No data for airport 10170 (code status: 400)


 56%|████████████████████████████████████████████▌                                   | 203/364 [02:38<02:02,  1.32it/s]

No data for airport 12448 (code status: 400)


 56%|█████████████████████████████████████████████                                   | 205/364 [02:40<01:59,  1.33it/s]

No data for airport 10918 (code status: 400)


 57%|█████████████████████████████████████████████▎                                  | 206/364 [02:40<01:59,  1.32it/s]

No data for airport 15356 (code status: 400)


 57%|█████████████████████████████████████████████▍                                  | 207/364 [02:41<01:58,  1.32it/s]

No data for airport 13344 (code status: 400)


 57%|█████████████████████████████████████████████▋                                  | 208/364 [02:42<01:58,  1.31it/s]

No data for airport 10676 (code status: 400)


 57%|█████████████████████████████████████████████▉                                  | 209/364 [02:43<01:58,  1.31it/s]

No data for airport 10926 (code status: 400)


 58%|██████████████████████████████████████████████▌                                 | 212/364 [02:45<01:55,  1.31it/s]

No data for airport 12156 (code status: 400)


 59%|██████████████████████████████████████████████▊                                 | 213/364 [02:46<01:54,  1.32it/s]

No data for airport 12244 (code status: 400)


 59%|███████████████████████████████████████████████                                 | 214/364 [02:46<01:54,  1.31it/s]

No data for airport 10666 (code status: 400)


 59%|███████████████████████████████████████████████▎                                | 215/364 [02:47<01:53,  1.31it/s]

No data for airport 11612 (code status: 400)


 59%|███████████████████████████████████████████████▍                                | 216/364 [02:48<01:52,  1.31it/s]

No data for airport 13476 (code status: 400)


 60%|███████████████████████████████████████████████▋                                | 217/364 [02:49<01:53,  1.30it/s]

No data for airport 13029 (code status: 400)


 60%|████████████████████████████████████████████████▏                               | 219/364 [02:50<01:50,  1.32it/s]

No data for airport 15411 (code status: 400)


 60%|████████████████████████████████████████████████▎                               | 220/364 [02:51<01:49,  1.31it/s]

No data for airport 14986 (code status: 400)


 61%|████████████████████████████████████████████████▌                               | 221/364 [02:52<01:49,  1.31it/s]

No data for airport 11471 (code status: 400)


 62%|█████████████████████████████████████████████████▏                              | 224/364 [02:54<01:51,  1.25it/s]

No data for airport 10155 (code status: 400)


 62%|█████████████████████████████████████████████████▍                              | 225/364 [02:55<01:49,  1.27it/s]

No data for airport 13486 (code status: 400)


 62%|█████████████████████████████████████████████████▋                              | 226/364 [02:56<01:48,  1.28it/s]

No data for airport 14716 (code status: 400)


 62%|█████████████████████████████████████████████████▉                              | 227/364 [02:57<01:46,  1.28it/s]

No data for airport 13296 (code status: 400)


 63%|██████████████████████████████████████████████████                              | 228/364 [02:57<01:45,  1.29it/s]

No data for airport 11283 (code status: 400)


 63%|██████████████████████████████████████████████████▎                             | 229/364 [02:58<01:42,  1.32it/s]

No data for airport 10732 (code status: 400)


 64%|██████████████████████████████████████████████████▉                             | 232/364 [03:00<01:39,  1.33it/s]

No data for airport 11003 (code status: 400)


 64%|███████████████████████████████████████████████████▏                            | 233/364 [03:01<01:39,  1.32it/s]

No data for airport 12094 (code status: 400)


 65%|███████████████████████████████████████████████████▋                            | 235/364 [03:03<01:37,  1.32it/s]

No data for airport 13158 (code status: 400)


 65%|███████████████████████████████████████████████████▊                            | 236/364 [03:03<01:36,  1.33it/s]

No data for airport 13121 (code status: 400)


 65%|████████████████████████████████████████████████████▎                           | 238/364 [03:05<01:34,  1.33it/s]

No data for airport 14877 (code status: 400)


 66%|████████████████████████████████████████████████████▌                           | 239/364 [03:06<01:37,  1.28it/s]

No data for airport 12217 (code status: 400)


 66%|████████████████████████████████████████████████████▉                           | 241/364 [03:07<01:44,  1.17it/s]

No data for airport 10372 (code status: 400)


 66%|█████████████████████████████████████████████████████▏                          | 242/364 [03:08<01:43,  1.18it/s]

No data for airport 13264 (code status: 400)


 67%|█████████████████████████████████████████████████████▍                          | 243/364 [03:09<01:43,  1.17it/s]

No data for airport 10469 (code status: 400)


 67%|█████████████████████████████████████████████████████▋                          | 244/364 [03:10<01:41,  1.19it/s]

No data for airport 11336 (code status: 400)


 68%|██████████████████████████████████████████████████████                          | 246/364 [03:12<01:40,  1.17it/s]

No data for airport 14112 (code status: 400)


 68%|██████████████████████████████████████████████████████▎                         | 247/364 [03:13<01:37,  1.20it/s]

No data for airport 13502 (code status: 400)


 68%|██████████████████████████████████████████████████████▋                         | 249/364 [03:14<01:35,  1.21it/s]

No data for airport 15027 (code status: 400)


 69%|██████████████████████████████████████████████████████▉                         | 250/364 [03:15<01:33,  1.22it/s]

No data for airport 11973 (code status: 400)


 69%|███████████████████████████████████████████████████████▍                        | 252/364 [03:17<01:29,  1.25it/s]

No data for airport 10361 (code status: 400)


 70%|███████████████████████████████████████████████████████▌                        | 253/364 [03:17<01:29,  1.24it/s]

No data for airport 13388 (code status: 400)


 70%|████████████████████████████████████████████████████████                        | 255/364 [03:19<01:26,  1.26it/s]

No data for airport 10577 (code status: 400)


 71%|████████████████████████████████████████████████████████▍                       | 257/364 [03:21<01:34,  1.14it/s]

No data for airport 13290 (code status: 400)


 71%|█████████████████████████████████████████████████████████▏                      | 260/364 [03:23<01:25,  1.22it/s]

No data for airport 11577 (code status: 400)


 72%|█████████████████████████████████████████████████████████▎                      | 261/364 [03:24<01:22,  1.25it/s]

No data for airport 11140 (code status: 400)


 72%|█████████████████████████████████████████████████████████▊                      | 263/364 [03:26<01:20,  1.25it/s]

No data for airport 15048 (code status: 400)


 73%|██████████████████████████████████████████████████████████                      | 264/364 [03:26<01:21,  1.23it/s]

No data for airport 14960 (code status: 400)


 73%|██████████████████████████████████████████████████████████▏                     | 265/364 [03:27<01:25,  1.16it/s]

No data for airport 15295 (code status: 400)


 73%|██████████████████████████████████████████████████████████▍                     | 266/364 [03:28<01:22,  1.19it/s]

No data for airport 11865 (code status: 400)


 74%|██████████████████████████████████████████████████████████▉                     | 268/364 [03:30<01:18,  1.22it/s]

No data for airport 11049 (code status: 400)


 74%|███████████████████████████████████████████████████████████▎                    | 270/364 [03:31<01:16,  1.23it/s]

No data for airport 11953 (code status: 400)


 74%|███████████████████████████████████████████████████████████▌                    | 271/364 [03:32<01:13,  1.26it/s]

No data for airport 12223 (code status: 400)


 75%|███████████████████████████████████████████████████████████▊                    | 272/364 [03:33<01:12,  1.26it/s]

No data for airport 14254 (code status: 400)


 75%|████████████████████████████████████████████████████████████                    | 273/364 [03:34<01:10,  1.29it/s]

No data for airport 11823 (code status: 400)


 75%|████████████████████████████████████████████████████████████▏                   | 274/364 [03:34<01:09,  1.29it/s]

No data for airport 15380 (code status: 400)


 76%|████████████████████████████████████████████████████████████▍                   | 275/364 [03:35<01:08,  1.29it/s]

No data for airport 11447 (code status: 400)


 76%|████████████████████████████████████████████████████████████▋                   | 276/364 [03:36<01:09,  1.27it/s]

No data for airport 12003 (code status: 400)


 76%|████████████████████████████████████████████████████████████▉                   | 277/364 [03:37<01:09,  1.25it/s]

No data for airport 14489 (code status: 400)


 77%|█████████████████████████████████████████████████████████████▌                  | 280/364 [03:39<01:10,  1.19it/s]

No data for airport 14082 (code status: 400)


 77%|█████████████████████████████████████████████████████████████▉                  | 282/364 [03:41<01:04,  1.26it/s]

No data for airport 14113 (code status: 400)


 78%|██████████████████████████████████████████████████████████████▏                 | 283/364 [03:42<01:03,  1.28it/s]

No data for airport 12891 (code status: 400)


 78%|██████████████████████████████████████████████████████████████▍                 | 284/364 [03:42<01:01,  1.30it/s]

No data for airport 15249 (code status: 400)


 78%|██████████████████████████████████████████████████████████████▋                 | 285/364 [03:43<01:00,  1.30it/s]

No data for airport 12819 (code status: 400)


 79%|███████████████████████████████████████████████████████████████                 | 287/364 [03:45<01:01,  1.26it/s]

No data for airport 12323 (code status: 400)


 79%|███████████████████████████████████████████████████████████████▌                | 289/364 [03:46<00:59,  1.26it/s]

No data for airport 12343 (code status: 400)


 80%|███████████████████████████████████████████████████████████████▋                | 290/364 [03:47<00:58,  1.26it/s]

No data for airport 13076 (code status: 400)


 80%|███████████████████████████████████████████████████████████████▉                | 291/364 [03:48<00:57,  1.26it/s]

No data for airport 11503 (code status: 400)


 80%|████████████████████████████████████████████████████████████████▏               | 292/364 [03:49<00:55,  1.29it/s]

No data for airport 11617 (code status: 400)


 80%|████████████████████████████████████████████████████████████████▍               | 293/364 [03:50<00:54,  1.29it/s]

No data for airport 10728 (code status: 400)


 81%|████████████████████████████████████████████████████████████████▌               | 294/364 [03:50<00:53,  1.30it/s]

No data for airport 10785 (code status: 400)


 82%|█████████████████████████████████████████████████████████████████▎              | 297/364 [03:53<00:50,  1.33it/s]

No data for airport 14674 (code status: 400)


 82%|█████████████████████████████████████████████████████████████████▍              | 298/364 [03:53<00:49,  1.34it/s]

No data for airport 12197 (code status: 400)


 82%|█████████████████████████████████████████████████████████████████▋              | 299/364 [03:54<00:49,  1.31it/s]

No data for airport 14696 (code status: 400)


 83%|██████████████████████████████████████████████████████████████████▏             | 301/364 [03:56<00:47,  1.33it/s]

No data for airport 10781 (code status: 400)


 83%|██████████████████████████████████████████████████████████████████▎             | 302/364 [03:56<00:46,  1.34it/s]

No data for airport 10849 (code status: 400)


 83%|██████████████████████████████████████████████████████████████████▌             | 303/364 [03:57<00:47,  1.29it/s]

No data for airport 14955 (code status: 400)


 84%|██████████████████████████████████████████████████████████████████▊             | 304/364 [03:58<00:46,  1.28it/s]

No data for airport 13433 (code status: 400)


 84%|███████████████████████████████████████████████████████████████████             | 305/364 [03:59<00:45,  1.29it/s]

No data for airport 15074 (code status: 400)


 84%|███████████████████████████████████████████████████████████████████▎            | 306/364 [03:59<00:44,  1.31it/s]

No data for airport 13127 (code status: 400)


 84%|███████████████████████████████████████████████████████████████████▍            | 307/364 [04:00<00:43,  1.32it/s]

No data for airport 11013 (code status: 400)


 85%|███████████████████████████████████████████████████████████████████▉            | 309/364 [04:02<00:42,  1.28it/s]

No data for airport 10154 (code status: 400)


 85%|████████████████████████████████████████████████████████████████████▏           | 310/364 [04:03<00:43,  1.26it/s]

No data for airport 10980 (code status: 400)


 85%|████████████████████████████████████████████████████████████████████▎           | 311/364 [04:03<00:42,  1.25it/s]

No data for airport 14098 (code status: 400)


 86%|████████████████████████████████████████████████████████████████████▊           | 313/364 [04:05<00:39,  1.29it/s]

No data for airport 14512 (code status: 400)


 86%|█████████████████████████████████████████████████████████████████████           | 314/364 [04:06<00:38,  1.29it/s]

No data for airport 12278 (code status: 400)


 87%|█████████████████████████████████████████████████████████████████████▉          | 318/364 [04:09<00:35,  1.31it/s]

No data for airport 13933 (code status: 400)


 88%|██████████████████████████████████████████████████████████████████████▎         | 320/364 [04:10<00:33,  1.33it/s]

No data for airport 11997 (code status: 400)


 88%|██████████████████████████████████████████████████████████████████████▌         | 321/364 [04:11<00:32,  1.33it/s]

No data for airport 11027 (code status: 400)


 88%|██████████████████████████████████████████████████████████████████████▊         | 322/364 [04:12<00:31,  1.32it/s]

No data for airport 15582 (code status: 400)


 89%|██████████████████████████████████████████████████████████████████████▉         | 323/364 [04:13<00:31,  1.30it/s]

No data for airport 11721 (code status: 400)


 89%|███████████████████████████████████████████████████████████████████████▏        | 324/364 [04:13<00:30,  1.32it/s]

No data for airport 12896 (code status: 400)


 89%|███████████████████████████████████████████████████████████████████████▍        | 325/364 [04:14<00:29,  1.32it/s]

No data for airport 10279 (code status: 400)


 90%|███████████████████████████████████████████████████████████████████████▋        | 326/364 [04:15<00:29,  1.30it/s]

No data for airport 11778 (code status: 400)


 90%|███████████████████████████████████████████████████████████████████████▊        | 327/364 [04:16<00:28,  1.30it/s]

No data for airport 12888 (code status: 400)


 90%|████████████████████████████████████████████████████████████████████████▎       | 329/364 [04:17<00:26,  1.32it/s]

No data for airport 11415 (code status: 400)


 91%|████████████████████████████████████████████████████████████████████████▌       | 330/364 [04:18<00:26,  1.29it/s]

No data for airport 10685 (code status: 400)


 91%|████████████████████████████████████████████████████████████████████████▋       | 331/364 [04:19<00:25,  1.30it/s]

No data for airport 14842 (code status: 400)


 92%|█████████████████████████████████████████████████████████████████████████▍      | 334/364 [04:21<00:22,  1.31it/s]

No data for airport 12265 (code status: 400)


 92%|█████████████████████████████████████████████████████████████████████████▋      | 335/364 [04:22<00:21,  1.32it/s]

No data for airport 14288 (code status: 400)


 93%|██████████████████████████████████████████████████████████████████████████      | 337/364 [04:23<00:20,  1.34it/s]

No data for airport 12397 (code status: 400)


 93%|██████████████████████████████████████████████████████████████████████████▎     | 338/364 [04:24<00:19,  1.31it/s]

No data for airport 15624 (code status: 400)


 93%|██████████████████████████████████████████████████████████████████████████▋     | 340/364 [04:25<00:18,  1.30it/s]

No data for airport 10409 (code status: 400)


 94%|██████████████████████████████████████████████████████████████████████████▉     | 341/364 [04:26<00:17,  1.31it/s]

No data for airport 12519 (code status: 400)


 94%|███████████████████████████████████████████████████████████████████████████▏    | 342/364 [04:27<00:16,  1.31it/s]

No data for airport 10136 (code status: 400)


 94%|███████████████████████████████████████████████████████████████████████████▍    | 343/364 [04:28<00:16,  1.30it/s]

No data for airport 15607 (code status: 400)


 95%|███████████████████████████████████████████████████████████████████████████▌    | 344/364 [04:29<00:15,  1.30it/s]

No data for airport 14237 (code status: 400)


 95%|███████████████████████████████████████████████████████████████████████████▊    | 345/364 [04:29<00:14,  1.30it/s]

No data for airport 10779 (code status: 400)


 95%|████████████████████████████████████████████████████████████████████████████    | 346/364 [04:30<00:13,  1.29it/s]

No data for airport 11898 (code status: 400)


 95%|████████████████████████████████████████████████████████████████████████████▎   | 347/364 [04:31<00:13,  1.29it/s]

No data for airport 11413 (code status: 400)


 96%|████████████████████████████████████████████████████████████████████████████▍   | 348/364 [04:32<00:12,  1.31it/s]

No data for airport 13983 (code status: 400)


 96%|████████████████████████████████████████████████████████████████████████████▋   | 349/364 [04:32<00:11,  1.32it/s]

No data for airport 10558 (code status: 400)


 96%|████████████████████████████████████████████████████████████████████████████▉   | 350/364 [04:33<00:10,  1.27it/s]

No data for airport 10620 (code status: 400)


 96%|█████████████████████████████████████████████████████████████████████████████▏  | 351/364 [04:34<00:10,  1.26it/s]

No data for airport 14222 (code status: 400)


 97%|█████████████████████████████████████████████████████████████████████████████▊  | 354/364 [04:36<00:07,  1.26it/s]

No data for airport 11097 (code status: 400)


 98%|██████████████████████████████████████████████████████████████████████████████  | 355/364 [04:37<00:07,  1.25it/s]

No data for airport 15454 (code status: 400)


 98%|██████████████████████████████████████████████████████████████████████████████▍ | 357/364 [04:39<00:05,  1.29it/s]

No data for airport 12899 (code status: 400)


 98%|██████████████████████████████████████████████████████████████████████████████▋ | 358/364 [04:39<00:04,  1.29it/s]

No data for airport 10135 (code status: 400)


 99%|██████████████████████████████████████████████████████████████████████████████▉ | 359/364 [04:40<00:03,  1.30it/s]

No data for airport 12177 (code status: 400)


 99%|███████████████████████████████████████████████████████████████████████████████▎| 361/364 [04:42<00:02,  1.28it/s]

No data for airport 13795 (code status: 400)


100%|███████████████████████████████████████████████████████████████████████████████▊| 363/364 [04:43<00:00,  1.28it/s]

No data for airport 12945 (code status: 400)


100%|████████████████████████████████████████████████████████████████████████████████| 364/364 [04:44<00:00,  1.28it/s]

No data for airport 12511 (code status: 400)


In [11]:
len(airports)

97

The output contains data for 97 airports out of 364. It is a list of dictionaries so **from.records()** function is used to create a Dataframe.

In [9]:
airport_df = pd.DataFrame.from_records(airports)

In [10]:
airport_df.head()

ORIGIN_AIRPORT_ID            DISPLAY_AIRPORT_NAME ORIGIN_CITY_NAME  \
0              11638             Fresno Air Terminal       Fresno, CA   
1              13342          General Mitchell Field    Milwaukee, WI   
2              13244           Memphis International      Memphis, TN   
3              15096  Syracuse Hancock International     Syracuse, NY   
4              10397               Atlanta Municipal      Atlanta, GA   

                                                NAME  
0               FRESNO YOSEMITE INTERNATIONAL, CA US  
1                  MILWAUKEE MITCHELL AIRPORT, WI US  
2               MEMPHIS INTERNATIONAL AIRPORT, TN US  
3      SYRACUSE HANCOCK INTERNATIONAL AIRPORT, NY US  
4  ATLANTA HARTSFIELD JACKSON INTERNATIONAL AIRPO...

In [12]:
#airport_df.to_csv(r'..\data\raw\airport_list.csv', index = False)

 # Collect data from `Weather` endpoint

In [13]:
months = ['2019-01', '2019-02', '2019-03', '2019-04', '2019-05', 
          '2019-06', '2019-07', '2019-08', '2019-09', '2019-10', 
          '2019-11', '2019-12', '2020-01', '2020-02', '2020-03']

dates=[]
for month in tqdm(months):
    response_dates = requests.get(f'{API_URL}/airportWeather?date={month}', headers={'Authorization': APIKEY})
    if response_dates.status_code == 200:    
        dates.append(response_dates.json())
    else:
        print(f'Error retrieving weather data {month} (code status: {response_airport.status_code})')

100%|██████████████████████████████████████████████████████████████████████████████████| 15/15 [00:26<00:00,  1.76s/it]


The output is a list of lists of dictionaries and needs to be flatten in order to create a Dataframe.

In [15]:
def flatten_data(API_content):
    flat_list = []
    for q in API_content:
        flat_list.extend(q)
    return flat_list

In [16]:
weather = flatten_data(dates)

In [19]:
weather_df = pd.DataFrame.from_records(weather)
weather_df.head()

WT18      STATION                                               NAME  \
0  None  USW00013874  ATLANTA HARTSFIELD JACKSON INTERNATIONAL AIRPO...   
1  None  USW00013874  ATLANTA HARTSFIELD JACKSON INTERNATIONAL AIRPO...   
2  None  USW00013874  ATLANTA HARTSFIELD JACKSON INTERNATIONAL AIRPO...   
3  None  USW00013874  ATLANTA HARTSFIELD JACKSON INTERNATIONAL AIRPO...   
4  None  USW00013874  ATLANTA HARTSFIELD JACKSON INTERNATIONAL AIRPO...   

         DATE   AWND  PRCP  SNOW  SNWD  TAVG  TMAX  ...  PGTM  WT10  WESD  \
0  2019-01-01   4.70  0.14   0.0   0.0  64.0  66.0  ...   NaN   NaN   NaN   
1  2019-01-02   4.92  0.57   0.0   0.0  56.0  59.0  ...   NaN   NaN   NaN   
2  2019-01-03   5.37  0.15   0.0   0.0  52.0  55.0  ...   NaN   NaN   NaN   
3  2019-01-04  12.08  1.44   0.0   0.0  56.0  66.0  ...   NaN   NaN   NaN   
4  2019-01-05  13.42  0.00   0.0   0.0  49.0  59.0  ...   NaN   NaN   NaN   

   SN32  SX32  PSUN  TSUN  TOBS  WT07  WT11  
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN  

[5 rows x 33 columns]

In [18]:
#weather_df.to_csv(r'..\data\raw\airport_weather.csv', index = False)

 # Collect data from `Aircraft` endpoint

In [21]:
response_aircraft = requests.get(f'{API_URL}/aircraft', headers={'Authorization': APIKEY})
response_aircraft.json()

[{'MANUFACTURE_YEAR': 1944, 'TAIL_NUM': 'N54514', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1945, 'TAIL_NUM': 'N1651M', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1953, 'TAIL_NUM': 'N100CE', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1953, 'TAIL_NUM': 'N141FL', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1953, 'TAIL_NUM': 'N151FL', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1953, 'TAIL_NUM': 'N381FL', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1954, 'TAIL_NUM': 'N6174C', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1954, 'TAIL_NUM': 'N400UA', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1955, 'TAIL_NUM': 'N747CE', 'NUMBER_OF_SEATS': 2},
 {'MANUFACTURE_YEAR': 1955, 'TAIL_NUM': 'N9148F', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1955, 'TAIL_NUM': 'N131FL', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1955, 'TAIL_NUM': 'N351FL', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1955, 'TAIL_NUM': 'N391FL', 'NUMBER_OF_SEATS': 0},
 {'MANUFACTURE_YEAR': 1956, 'TAIL_NUM': 'N171FL', '

In [22]:
aircraft_df = pd.DataFrame.from_records(response_aircraft.json())
aircraft_df.head()

MANUFACTURE_YEAR TAIL_NUM  NUMBER_OF_SEATS
0              1944   N54514              0.0
1              1945   N1651M              0.0
2              1953   N100CE              0.0
3              1953   N141FL              0.0
4              1953   N151FL              0.0

In [22]:
#aircraft_df.to_csv(r'..\data\raw\aircraft.csv', index = False)

 # Collect data from `Flight` endpoint

In [26]:
flights_data=[]

def download_airport_monthly_data(airport_id, month):
    response_flights = requests.get(f'{API_URL}/flight?airportId={airport_id}&date={month}', headers={'Authorization': APIKEY})
    
    if response_flights.status_code != 204:  
   
        print(f'getting date {month} from airport {airport_id}')

    elif response_flights.status_code == 204:
        return None

    else:
        print(f"AirportId: {airport_id} | Month: {month} | Status code: " + int(response_flights.status_code))

    result = response_flights.json() 
    return result

def download_airport_data(airport):
    result = []
    for month in months:
        sleep(60/1000)
        data = download_airport_monthly_data(airport, month)

        if data is None:
            break

        result += data
    return result

for airport in tqdm(airport_id_list):
    flights_data += download_airport_data(airport)

  5%|████                                                                             | 18/364 [00:33<10:41,  1.85s/it]

getting date 2019-01 from airport 10397
getting date 2019-02 from airport 10397
getting date 2019-03 from airport 10397
getting date 2019-04 from airport 10397
getting date 2019-05 from airport 10397
getting date 2019-06 from airport 10397
getting date 2019-07 from airport 10397
getting date 2019-08 from airport 10397
getting date 2019-09 from airport 10397
getting date 2019-10 from airport 10397
getting date 2019-11 from airport 10397
getting date 2019-12 from airport 10397
getting date 2020-01 from airport 10397
getting date 2020-02 from airport 10397


  5%|████                                                                           | 19/364 [02:02<2:40:50, 27.97s/it]

getting date 2020-03 from airport 10397


 11%|█████████                                                                        | 41/364 [02:42<09:58,  1.85s/it]

getting date 2019-01 from airport 14831
getting date 2019-02 from airport 14831
getting date 2019-03 from airport 14831
getting date 2019-04 from airport 14831
getting date 2019-05 from airport 14831
getting date 2019-06 from airport 14831
getting date 2019-07 from airport 14831
getting date 2019-08 from airport 14831
getting date 2019-09 from airport 14831
getting date 2019-10 from airport 14831
getting date 2019-11 from airport 14831
getting date 2019-12 from airport 14831
getting date 2020-01 from airport 14831
getting date 2020-02 from airport 14831


 12%|█████████                                                                      | 42/364 [03:15<1:00:05, 11.20s/it]

getting date 2020-03 from airport 14831


 15%|███████████▊                                                                     | 53/364 [03:35<10:14,  1.98s/it]

getting date 2019-01 from airport 11292
getting date 2019-02 from airport 11292
getting date 2019-03 from airport 11292
getting date 2019-04 from airport 11292
getting date 2019-05 from airport 11292
getting date 2019-06 from airport 11292
getting date 2019-07 from airport 11292
getting date 2019-08 from airport 11292
getting date 2019-09 from airport 11292
getting date 2019-10 from airport 11292
getting date 2019-11 from airport 11292
getting date 2019-12 from airport 11292
getting date 2020-01 from airport 11292
getting date 2020-02 from airport 11292


 15%|███████████▋                                                                   | 54/364 [04:50<2:03:45, 23.95s/it]

getting date 2020-03 from airport 11292


 17%|██████████████                                                                   | 63/364 [05:07<13:40,  2.73s/it]

getting date 2019-01 from airport 11259
getting date 2019-02 from airport 11259
getting date 2019-03 from airport 11259
getting date 2019-04 from airport 11259
getting date 2019-05 from airport 11259
getting date 2019-06 from airport 11259
getting date 2019-07 from airport 11259
getting date 2019-08 from airport 11259
getting date 2019-09 from airport 11259
getting date 2019-10 from airport 11259
getting date 2019-11 from airport 11259
getting date 2019-12 from airport 11259
getting date 2020-01 from airport 11259
getting date 2020-02 from airport 11259


 18%|██████████████▏                                                                  | 64/364 [05:40<59:13, 11.85s/it]

getting date 2020-03 from airport 11259


 18%|██████████████▋                                                                  | 66/364 [05:44<33:37,  6.77s/it]

getting date 2019-01 from airport 12402
getting date 2019-02 from airport 12402
getting date 2019-03 from airport 12402
getting date 2019-04 from airport 12402
getting date 2019-05 from airport 12402
getting date 2019-06 from airport 12402
getting date 2019-07 from airport 12402
getting date 2019-08 from airport 12402
getting date 2019-09 from airport 12402
getting date 2019-10 from airport 12402
getting date 2019-11 from airport 12402
getting date 2019-12 from airport 12402
getting date 2020-01 from airport 12402
getting date 2020-02 from airport 12402


 18%|██████████████▌                                                                | 67/364 [06:16<1:11:25, 14.43s/it]

getting date 2020-03 from airport 12402


 23%|██████████████████▋                                                              | 84/364 [06:47<08:55,  1.91s/it]

getting date 2019-01 from airport 11618
getting date 2019-02 from airport 11618
getting date 2019-03 from airport 11618
getting date 2019-04 from airport 11618
getting date 2019-05 from airport 11618
getting date 2019-06 from airport 11618
getting date 2019-07 from airport 11618
getting date 2019-08 from airport 11618
getting date 2019-09 from airport 11618
getting date 2019-10 from airport 11618
getting date 2019-11 from airport 11618
getting date 2019-12 from airport 11618
getting date 2020-01 from airport 11618
getting date 2020-02 from airport 11618


 23%|██████████████████▍                                                            | 85/364 [07:33<1:10:39, 15.20s/it]

getting date 2020-03 from airport 11618
getting date 2019-01 from airport 15304
getting date 2019-02 from airport 15304
getting date 2019-03 from airport 15304
getting date 2019-04 from airport 15304
getting date 2019-05 from airport 15304
getting date 2019-06 from airport 15304
getting date 2019-07 from airport 15304
getting date 2019-08 from airport 15304
getting date 2019-09 from airport 15304
getting date 2019-10 from airport 15304
getting date 2019-11 from airport 15304
getting date 2019-12 from airport 15304
getting date 2020-01 from airport 15304
getting date 2020-02 from airport 15304


 24%|██████████████████▋                                                            | 86/364 [08:04<1:32:29, 19.96s/it]

getting date 2020-03 from airport 15304


 36%|████████████████████████████▊                                                   | 131/364 [09:27<06:57,  1.79s/it]

getting date 2019-01 from airport 14107
getting date 2019-02 from airport 14107
getting date 2019-03 from airport 14107
getting date 2019-04 from airport 14107
getting date 2019-05 from airport 14107
getting date 2019-06 from airport 14107
getting date 2019-07 from airport 14107
getting date 2019-08 from airport 14107
getting date 2019-09 from airport 14107
getting date 2019-10 from airport 14107
getting date 2019-11 from airport 14107
getting date 2019-12 from airport 14107
getting date 2020-01 from airport 14107
getting date 2020-02 from airport 14107


 36%|████████████████████████████▎                                                 | 132/364 [10:30<1:18:26, 20.29s/it]

getting date 2020-03 from airport 14107


 37%|█████████████████████████████▏                                                  | 133/364 [10:32<56:50, 14.77s/it]

getting date 2019-01 from airport 14747
getting date 2019-02 from airport 14747
getting date 2019-03 from airport 14747
getting date 2019-04 from airport 14747
getting date 2019-05 from airport 14747
getting date 2019-06 from airport 14747
getting date 2019-07 from airport 14747
getting date 2019-08 from airport 14747
getting date 2019-09 from airport 14747
getting date 2019-10 from airport 14747
getting date 2019-11 from airport 14747
getting date 2019-12 from airport 14747
getting date 2020-01 from airport 14747
getting date 2020-02 from airport 14747


 37%|████████████████████████████▋                                                 | 134/364 [11:29<1:44:43, 27.32s/it]

getting date 2020-03 from airport 14747


 39%|██████████████████████████████▉                                                 | 141/364 [11:42<14:41,  3.95s/it]

getting date 2019-01 from airport 12889
getting date 2019-02 from airport 12889
getting date 2019-03 from airport 12889
getting date 2019-04 from airport 12889
getting date 2019-05 from airport 12889
getting date 2019-06 from airport 12889
getting date 2019-07 from airport 12889
getting date 2019-08 from airport 12889
getting date 2019-09 from airport 12889
getting date 2019-10 from airport 12889
getting date 2019-11 from airport 12889
getting date 2019-12 from airport 12889
getting date 2020-01 from airport 12889
getting date 2020-02 from airport 12889


 39%|██████████████████████████████▍                                               | 142/364 [12:40<1:15:05, 20.30s/it]

getting date 2020-03 from airport 12889


 40%|████████████████████████████████                                                | 146/364 [12:47<22:40,  6.24s/it]

getting date 2019-01 from airport 14492
getting date 2019-02 from airport 14492
getting date 2019-03 from airport 14492
getting date 2019-04 from airport 14492
getting date 2019-05 from airport 14492
getting date 2019-06 from airport 14492
getting date 2019-07 from airport 14492
getting date 2019-08 from airport 14492
getting date 2019-09 from airport 14492
getting date 2019-10 from airport 14492
getting date 2019-11 from airport 14492
getting date 2019-12 from airport 14492
getting date 2020-01 from airport 14492
getting date 2020-02 from airport 14492


 40%|████████████████████████████████▎                                               | 147/364 [13:22<53:32, 14.80s/it]

getting date 2020-03 from airport 14492


 41%|████████████████████████████████▌                                               | 148/364 [13:24<39:20, 10.93s/it]

getting date 2019-01 from airport 13204
getting date 2019-02 from airport 13204
getting date 2019-03 from airport 13204
getting date 2019-04 from airport 13204
getting date 2019-05 from airport 13204
getting date 2019-06 from airport 13204
getting date 2019-07 from airport 13204
getting date 2019-08 from airport 13204
getting date 2019-09 from airport 13204
getting date 2019-10 from airport 13204
getting date 2019-11 from airport 13204
getting date 2019-12 from airport 13204
getting date 2020-01 from airport 13204
getting date 2020-02 from airport 13204


 41%|███████████████████████████████▉                                              | 149/364 [14:08<1:15:05, 20.96s/it]

getting date 2020-03 from airport 13204


 44%|███████████████████████████████████▍                                            | 161/364 [14:31<07:11,  2.12s/it]

getting date 2019-01 from airport 13487
getting date 2019-02 from airport 13487
getting date 2019-03 from airport 13487
getting date 2019-04 from airport 13487
getting date 2019-05 from airport 13487
getting date 2019-06 from airport 13487
getting date 2019-07 from airport 13487
getting date 2019-08 from airport 13487
getting date 2019-09 from airport 13487
getting date 2019-10 from airport 13487
getting date 2019-11 from airport 13487
getting date 2019-12 from airport 13487
getting date 2020-01 from airport 13487
getting date 2020-02 from airport 13487


 45%|███████████████████████████████████▌                                            | 162/364 [15:11<46:08, 13.71s/it]

getting date 2020-03 from airport 13487


 46%|████████████████████████████████████▋                                           | 167/364 [15:21<12:40,  3.86s/it]

getting date 2019-01 from airport 10299
getting date 2019-02 from airport 10299
getting date 2019-03 from airport 10299
getting date 2019-04 from airport 10299
getting date 2019-05 from airport 10299
getting date 2019-06 from airport 10299
getting date 2019-07 from airport 10299
getting date 2019-08 from airport 10299
getting date 2019-09 from airport 10299
getting date 2019-10 from airport 10299
getting date 2019-11 from airport 10299
getting date 2019-12 from airport 10299
getting date 2020-01 from airport 10299
getting date 2020-02 from airport 10299


 46%|████████████████████████████████████▉                                           | 168/364 [16:00<46:49, 14.33s/it]

getting date 2020-03 from airport 10299


 52%|█████████████████████████████████████████▌                                      | 189/364 [16:38<05:28,  1.88s/it]

getting date 2019-01 from airport 11278
getting date 2019-02 from airport 11278
getting date 2019-03 from airport 11278
getting date 2019-04 from airport 11278
getting date 2019-05 from airport 11278
getting date 2019-06 from airport 11278
getting date 2019-07 from airport 11278
getting date 2019-08 from airport 11278
getting date 2019-09 from airport 11278
getting date 2019-10 from airport 11278
getting date 2019-11 from airport 11278
getting date 2019-12 from airport 11278
getting date 2020-01 from airport 11278
getting date 2020-02 from airport 11278


 52%|█████████████████████████████████████████▊                                      | 190/364 [17:27<45:53, 15.82s/it]

getting date 2020-03 from airport 11278


 52%|█████████████████████████████████████████▉                                      | 191/364 [17:29<33:27, 11.60s/it]

getting date 2019-01 from airport 14100
getting date 2019-02 from airport 14100
getting date 2019-03 from airport 14100
getting date 2019-04 from airport 14100
getting date 2019-05 from airport 14100
getting date 2019-06 from airport 14100
getting date 2019-07 from airport 14100
getting date 2019-08 from airport 14100
getting date 2019-09 from airport 14100
getting date 2019-10 from airport 14100
getting date 2019-11 from airport 14100
getting date 2019-12 from airport 14100
getting date 2020-01 from airport 14100
getting date 2020-02 from airport 14100


 53%|██████████████████████████████████████████▏                                     | 192/364 [18:11<59:41, 20.82s/it]

getting date 2020-03 from airport 14100


 54%|███████████████████████████████████████████▌                                    | 198/364 [18:22<11:05,  4.01s/it]

getting date 2019-01 from airport 10821
getting date 2019-02 from airport 10821
getting date 2019-03 from airport 10821
getting date 2019-04 from airport 10821
getting date 2019-05 from airport 10821
getting date 2019-06 from airport 10821
getting date 2019-07 from airport 10821
getting date 2019-08 from airport 10821
getting date 2019-09 from airport 10821
getting date 2019-10 from airport 10821
getting date 2019-11 from airport 10821
getting date 2019-12 from airport 10821
getting date 2020-01 from airport 10821
getting date 2020-02 from airport 10821


 55%|███████████████████████████████████████████▋                                    | 199/364 [19:00<39:42, 14.44s/it]

getting date 2020-03 from airport 10821
getting date 2019-01 from airport 10721
getting date 2019-02 from airport 10721
getting date 2019-03 from airport 10721
getting date 2019-04 from airport 10721
getting date 2019-05 from airport 10721
getting date 2019-06 from airport 10721
getting date 2019-07 from airport 10721
getting date 2019-08 from airport 10721
getting date 2019-09 from airport 10721
getting date 2019-10 from airport 10721
getting date 2019-11 from airport 10721
getting date 2019-12 from airport 10721
getting date 2020-01 from airport 10721
getting date 2020-02 from airport 10721


 55%|██████████████████████████████████████████▊                                   | 200/364 [19:53<1:10:52, 25.93s/it]

getting date 2020-03 from airport 10721


 56%|████████████████████████████████████████████▌                                   | 203/364 [19:59<27:20, 10.19s/it]

getting date 2019-01 from airport 12953
getting date 2019-02 from airport 12953
getting date 2019-03 from airport 12953
getting date 2019-04 from airport 12953
getting date 2019-05 from airport 12953
getting date 2019-06 from airport 12953
getting date 2019-07 from airport 12953
getting date 2019-08 from airport 12953
getting date 2019-09 from airport 12953
getting date 2019-10 from airport 12953
getting date 2019-11 from airport 12953
getting date 2019-12 from airport 12953
getting date 2020-01 from airport 12953
getting date 2020-02 from airport 12953


 56%|███████████████████████████████████████████▋                                  | 204/364 [20:57<1:05:42, 24.64s/it]

getting date 2020-03 from airport 12953


 61%|████████████████████████████████████████████████▌                               | 221/364 [21:30<04:43,  1.98s/it]

getting date 2019-01 from airport 12758
getting date 2019-02 from airport 12758
getting date 2019-03 from airport 12758
getting date 2019-04 from airport 12758
getting date 2019-05 from airport 12758
getting date 2019-06 from airport 12758
getting date 2019-07 from airport 12758
getting date 2019-08 from airport 12758
getting date 2019-09 from airport 12758
getting date 2019-10 from airport 12758
getting date 2019-11 from airport 12758
getting date 2019-12 from airport 12758
getting date 2020-01 from airport 12758
getting date 2020-02 from airport 12758


 61%|████████████████████████████████████████████████▊                               | 222/364 [22:05<28:36, 12.09s/it]

getting date 2020-03 from airport 12758


 63%|██████████████████████████████████████████████████▎                             | 229/364 [22:18<06:05,  2.70s/it]

getting date 2019-01 from airport 12266
getting date 2019-02 from airport 12266
getting date 2019-03 from airport 12266
getting date 2019-04 from airport 12266
getting date 2019-05 from airport 12266
getting date 2019-06 from airport 12266
getting date 2019-07 from airport 12266
getting date 2019-08 from airport 12266
getting date 2019-09 from airport 12266
getting date 2019-10 from airport 12266
getting date 2019-11 from airport 12266
getting date 2019-12 from airport 12266
getting date 2020-01 from airport 12266
getting date 2020-02 from airport 12266


 63%|██████████████████████████████████████████████████▌                             | 230/364 [22:55<28:58, 12.97s/it]

getting date 2020-03 from airport 12266
getting date 2019-01 from airport 12892
getting date 2019-02 from airport 12892
getting date 2019-03 from airport 12892
getting date 2019-04 from airport 12892
getting date 2019-05 from airport 12892
getting date 2019-06 from airport 12892
getting date 2019-07 from airport 12892
getting date 2019-08 from airport 12892
getting date 2019-09 from airport 12892
getting date 2019-10 from airport 12892
getting date 2019-11 from airport 12892
getting date 2019-12 from airport 12892
getting date 2020-01 from airport 12892
getting date 2020-02 from airport 12892


 63%|█████████████████████████████████████████████████▌                            | 231/364 [24:23<1:18:38, 35.47s/it]

getting date 2020-03 from airport 12892


 66%|████████████████████████████████████████████████████▌                           | 239/364 [24:38<07:55,  3.80s/it]

getting date 2019-01 from airport 11697
getting date 2019-02 from airport 11697
getting date 2019-03 from airport 11697
getting date 2019-04 from airport 11697
getting date 2019-05 from airport 11697
getting date 2019-06 from airport 11697
getting date 2019-07 from airport 11697
getting date 2019-08 from airport 11697
getting date 2019-09 from airport 11697
getting date 2019-10 from airport 11697
getting date 2019-11 from airport 11697
getting date 2019-12 from airport 11697
getting date 2020-01 from airport 11697
getting date 2020-02 from airport 11697


 66%|████████████████████████████████████████████████████▋                           | 240/364 [25:14<27:26, 13.28s/it]

getting date 2020-03 from airport 11697


 69%|██████████████████████████████████████████████████████▉                         | 250/364 [25:32<04:16,  2.25s/it]

getting date 2019-01 from airport 12173
getting date 2019-02 from airport 12173
getting date 2019-03 from airport 12173
getting date 2019-04 from airport 12173
getting date 2019-05 from airport 12173
getting date 2019-06 from airport 12173
getting date 2019-07 from airport 12173
getting date 2019-08 from airport 12173
getting date 2019-09 from airport 12173
getting date 2019-10 from airport 12173
getting date 2019-11 from airport 12173
getting date 2019-12 from airport 12173
getting date 2020-01 from airport 12173
getting date 2020-02 from airport 12173


 69%|███████████████████████████████████████████████████████▏                        | 251/364 [26:30<35:16, 18.73s/it]

getting date 2020-03 from airport 12173


 70%|████████████████████████████████████████████████████████                        | 255/364 [26:37<10:46,  5.93s/it]

getting date 2019-01 from airport 11433
getting date 2019-02 from airport 11433
getting date 2019-03 from airport 11433
getting date 2019-04 from airport 11433
getting date 2019-05 from airport 11433
getting date 2019-06 from airport 11433
getting date 2019-07 from airport 11433
getting date 2019-08 from airport 11433
getting date 2019-09 from airport 11433
getting date 2019-10 from airport 11433
getting date 2019-11 from airport 11433
getting date 2019-12 from airport 11433
getting date 2020-01 from airport 11433
getting date 2020-02 from airport 11433


 70%|████████████████████████████████████████████████████████▎                       | 256/364 [27:10<25:24, 14.12s/it]

getting date 2020-03 from airport 11433


 71%|████████████████████████████████████████████████████████▍                       | 257/364 [27:12<18:37, 10.44s/it]

getting date 2019-01 from airport 11298
getting date 2019-02 from airport 11298
getting date 2019-03 from airport 11298
getting date 2019-04 from airport 11298
getting date 2019-05 from airport 11298
getting date 2019-06 from airport 11298
getting date 2019-07 from airport 11298
getting date 2019-08 from airport 11298
getting date 2019-09 from airport 11298
getting date 2019-10 from airport 11298
getting date 2019-11 from airport 11298
getting date 2019-12 from airport 11298
getting date 2020-01 from airport 11298
getting date 2020-02 from airport 11298


 71%|████████████████████████████████████████████████████████▋                       | 258/364 [28:06<41:29, 23.49s/it]

getting date 2020-03 from airport 11298
getting date 2019-01 from airport 14771
getting date 2019-02 from airport 14771
getting date 2019-03 from airport 14771
getting date 2019-04 from airport 14771
getting date 2019-05 from airport 14771
getting date 2019-06 from airport 14771
getting date 2019-07 from airport 14771
getting date 2019-08 from airport 14771
getting date 2019-09 from airport 14771
getting date 2019-10 from airport 14771
getting date 2019-11 from airport 14771
getting date 2019-12 from airport 14771
getting date 2020-01 from airport 14771
getting date 2020-02 from airport 14771


 71%|████████████████████████████████████████████████████████▉                       | 259/364 [28:58<56:07, 32.07s/it]

getting date 2020-03 from airport 14771


 72%|█████████████████████████████████████████████████████████▎                      | 261/364 [29:02<28:35, 16.66s/it]

getting date 2019-01 from airport 11057
getting date 2019-02 from airport 11057
getting date 2019-03 from airport 11057
getting date 2019-04 from airport 11057
getting date 2019-05 from airport 11057
getting date 2019-06 from airport 11057
getting date 2019-07 from airport 11057
getting date 2019-08 from airport 11057
getting date 2019-09 from airport 11057
getting date 2019-10 from airport 11057
getting date 2019-11 from airport 11057
getting date 2019-12 from airport 11057
getting date 2020-01 from airport 11057
getting date 2020-02 from airport 11057


 72%|█████████████████████████████████████████████████████████▌                      | 262/364 [29:39<38:42, 22.77s/it]

getting date 2020-03 from airport 11057


 73%|██████████████████████████████████████████████████████████▍                     | 266/364 [29:46<11:12,  6.86s/it]

getting date 2019-01 from airport 14893
getting date 2019-02 from airport 14893
getting date 2019-03 from airport 14893
getting date 2019-04 from airport 14893
getting date 2019-05 from airport 14893
getting date 2019-06 from airport 14893
getting date 2019-07 from airport 14893
getting date 2019-08 from airport 14893
getting date 2019-09 from airport 14893
getting date 2019-10 from airport 14893
getting date 2019-11 from airport 14893
getting date 2019-12 from airport 14893
getting date 2020-01 from airport 14893
getting date 2020-02 from airport 14893


 73%|██████████████████████████████████████████████████████████▋                     | 267/364 [30:20<23:56, 14.81s/it]

getting date 2020-03 from airport 14893


 74%|██████████████████████████████████████████████████████████▉                     | 268/364 [30:22<17:29, 10.93s/it]

getting date 2019-01 from airport 13495
getting date 2019-02 from airport 13495
getting date 2019-03 from airport 13495
getting date 2019-04 from airport 13495
getting date 2019-05 from airport 13495
getting date 2019-06 from airport 13495
getting date 2019-07 from airport 13495
getting date 2019-08 from airport 13495
getting date 2019-09 from airport 13495
getting date 2019-10 from airport 13495
getting date 2019-11 from airport 13495
getting date 2019-12 from airport 13495
getting date 2020-01 from airport 13495
getting date 2020-02 from airport 13495


 74%|███████████████████████████████████████████████████████████                     | 269/364 [30:56<28:25, 17.95s/it]

getting date 2020-03 from airport 13495


 77%|█████████████████████████████████████████████████████████████▌                  | 280/364 [31:17<03:05,  2.21s/it]

getting date 2019-01 from airport 12191
getting date 2019-02 from airport 12191
getting date 2019-03 from airport 12191
getting date 2019-04 from airport 12191
getting date 2019-05 from airport 12191
getting date 2019-06 from airport 12191
getting date 2019-07 from airport 12191
getting date 2019-08 from airport 12191
getting date 2019-09 from airport 12191
getting date 2019-10 from airport 12191
getting date 2019-11 from airport 12191
getting date 2019-12 from airport 12191
getting date 2020-01 from airport 12191
getting date 2020-02 from airport 12191


 77%|█████████████████████████████████████████████████████████████▊                  | 281/364 [31:52<16:39, 12.04s/it]

getting date 2020-03 from airport 12191


 79%|███████████████████████████████████████████████████████████████                 | 287/364 [32:04<04:00,  3.13s/it]

getting date 2019-01 from airport 14679
getting date 2019-02 from airport 14679
getting date 2019-03 from airport 14679
getting date 2019-04 from airport 14679
getting date 2019-05 from airport 14679
getting date 2019-06 from airport 14679
getting date 2019-07 from airport 14679
getting date 2019-08 from airport 14679
getting date 2019-09 from airport 14679
getting date 2019-10 from airport 14679
getting date 2019-11 from airport 14679
getting date 2019-12 from airport 14679
getting date 2020-01 from airport 14679
getting date 2020-02 from airport 14679


 79%|███████████████████████████████████████████████████████████████▎                | 288/364 [32:44<18:11, 14.36s/it]

getting date 2020-03 from airport 14679


 81%|████████████████████████████████████████████████████████████████▌               | 294/364 [32:56<03:56,  3.38s/it]

getting date 2019-01 from airport 14869
getting date 2019-02 from airport 14869
getting date 2019-03 from airport 14869
getting date 2019-04 from airport 14869
getting date 2019-05 from airport 14869
getting date 2019-06 from airport 14869
getting date 2019-07 from airport 14869
getting date 2019-08 from airport 14869
getting date 2019-09 from airport 14869
getting date 2019-10 from airport 14869
getting date 2019-11 from airport 14869
getting date 2019-12 from airport 14869
getting date 2020-01 from airport 14869
getting date 2020-02 from airport 14869


 81%|████████████████████████████████████████████████████████████████▊               | 295/364 [33:36<16:41, 14.52s/it]

getting date 2020-03 from airport 14869


 85%|████████████████████████████████████████████████████████████████████▎           | 311/364 [34:06<01:42,  1.93s/it]

getting date 2019-01 from airport 12982
getting date 2019-02 from airport 12982
getting date 2019-03 from airport 12982
getting date 2019-04 from airport 12982
getting date 2019-05 from airport 12982
getting date 2019-06 from airport 12982
getting date 2019-07 from airport 12982
getting date 2019-08 from airport 12982
getting date 2019-09 from airport 12982
getting date 2019-10 from airport 12982
getting date 2019-11 from airport 12982
getting date 2019-12 from airport 12982
getting date 2020-01 from airport 12982
getting date 2020-02 from airport 12982


 86%|████████████████████████████████████████████████████████████████████▌           | 312/364 [34:41<10:14, 11.82s/it]

getting date 2020-03 from airport 12982


 87%|█████████████████████████████████████████████████████████████████████▏          | 315/364 [34:47<04:18,  5.28s/it]

getting date 2019-01 from airport 12478
getting date 2019-02 from airport 12478
getting date 2019-03 from airport 12478
getting date 2019-04 from airport 12478
getting date 2019-05 from airport 12478
getting date 2019-06 from airport 12478
getting date 2019-07 from airport 12478
getting date 2019-08 from airport 12478
getting date 2019-09 from airport 12478
getting date 2019-10 from airport 12478
getting date 2019-11 from airport 12478
getting date 2019-12 from airport 12478
getting date 2020-01 from airport 12478
getting date 2020-02 from airport 12478


 87%|█████████████████████████████████████████████████████████████████████▍          | 316/364 [35:31<13:41, 17.11s/it]

getting date 2020-03 from airport 12478
getting date 2019-01 from airport 13830
getting date 2019-02 from airport 13830
getting date 2019-03 from airport 13830
getting date 2019-04 from airport 13830
getting date 2019-05 from airport 13830
getting date 2019-06 from airport 13830
getting date 2019-07 from airport 13830
getting date 2019-08 from airport 13830
getting date 2019-09 from airport 13830
getting date 2019-10 from airport 13830
getting date 2019-11 from airport 13830
getting date 2019-12 from airport 13830
getting date 2020-01 from airport 13830
getting date 2020-02 from airport 13830


 87%|█████████████████████████████████████████████████████████████████████▋          | 317/364 [36:11<18:35, 23.74s/it]

getting date 2020-03 from airport 13830


 87%|█████████████████████████████████████████████████████████████████████▉          | 318/364 [36:12<13:10, 17.18s/it]

getting date 2019-01 from airport 13930
getting date 2019-02 from airport 13930
getting date 2019-03 from airport 13930
getting date 2019-04 from airport 13930
getting date 2019-05 from airport 13930
getting date 2019-06 from airport 13930
getting date 2019-07 from airport 13930
getting date 2019-08 from airport 13930
getting date 2019-09 from airport 13930
getting date 2019-10 from airport 13930
getting date 2019-11 from airport 13930
getting date 2019-12 from airport 13930
getting date 2020-01 from airport 13930
getting date 2020-02 from airport 13930


 88%|██████████████████████████████████████████████████████████████████████          | 319/364 [37:40<28:49, 38.44s/it]

getting date 2020-03 from airport 13930


 91%|████████████████████████████████████████████████████████████████████████▉       | 332/364 [38:05<01:12,  2.26s/it]

getting date 2019-01 from airport 11193
getting date 2019-02 from airport 11193
getting date 2019-03 from airport 11193
getting date 2019-04 from airport 11193
getting date 2019-05 from airport 11193
getting date 2019-06 from airport 11193
getting date 2019-07 from airport 11193
getting date 2019-08 from airport 11193
getting date 2019-09 from airport 11193
getting date 2019-10 from airport 11193
getting date 2019-11 from airport 11193
getting date 2019-12 from airport 11193
getting date 2020-01 from airport 11193
getting date 2020-02 from airport 11193


 91%|█████████████████████████████████████████████████████████████████████████▏      | 333/364 [38:37<05:48, 11.25s/it]

getting date 2020-03 from airport 11193


 99%|██████████████████████████████████████████████████████████████████████████████▉ | 359/364 [39:26<00:09,  1.88s/it]

getting date 2019-01 from airport 13303
getting date 2019-02 from airport 13303
getting date 2019-03 from airport 13303
getting date 2019-04 from airport 13303
getting date 2019-05 from airport 13303
getting date 2019-06 from airport 13303
getting date 2019-07 from airport 13303
getting date 2019-08 from airport 13303
getting date 2019-09 from airport 13303
getting date 2019-10 from airport 13303
getting date 2019-11 from airport 13303
getting date 2019-12 from airport 13303
getting date 2020-01 from airport 13303
getting date 2020-02 from airport 13303


 99%|███████████████████████████████████████████████████████████████████████████████ | 360/364 [40:05<00:51, 12.77s/it]

getting date 2020-03 from airport 13303


100%|████████████████████████████████████████████████████████████████████████████████| 364/364 [40:12<00:00,  6.63s/it]


In [27]:
len(flights_data)
    

1386120

In [28]:
flight_df =pd.json_normalize(flights_data)

In [30]:
flight_df.shape

(1386120, 27)

In [32]:
# flight_df.to_csv(r'..\data\raw\flight.csv', index = False)

 # Summary
 
The data from all available endpoints was collected successfully and saved to csv files. It can be accessed in project repository in data/raw/ folder.
When collecting data from the Flight endpoint, a SLEEP function was employed in order to not exceed the API call limit. Also the 'tqdm' function was used to monitor the loops progress.